improve esm docs (#6232)
webpack/webpack.js.org: src/content/guides/ecma-script-modules.mdx

In addition to the module format, flagging modules as ESM also affect the resolving logic, interop logic and the available symbols in modules.

> Imports in ESM are resolved more strictly. Relative requests must include a filename and file extension (e.g. `*.js` or `*.mjs`) to disable this behaviour set [`fullySpecified=false`](/configuration/module/#resolvefullyspecified).

Reviewer: ```suggestion
Imports in ESM are resolved more strictly. Relative requests must include a filename and file extension (e.g. `*.js` or `*.mjs`) unless you have the behaviour disabled with [`fullySpecified=false`](/configuration/module/#resolvefullyspecified).
```
